src/libs/data-utils.js

```diff
@@ -62,107 +62,126 @@ class UriViewer extends Component {
     const isGsUri = isGs(uri)
     const [bucket, name] = isGsUri ? parseUri(uri) : []
 
-    const { signedUrl = false, ...metadata } = isGsUri ? await Buckets.getObject(bucket, name, googleProject) : await Martha.call(uri)
+    try {
+      const { signedUrl = false, ...metadata } = isGsUri ? await Buckets.getObject(bucket, name, googleProject) : await Martha.call(uri)
 
-    const price = getMaxDownloadCostNA(metadata.size)
+      const price = getMaxDownloadCostNA(metadata.size)
 
-    this.setState(_.merge({ metadata, price }, !isGsUri && { signedUrl }))
+      this.setState(_.merge({
```

Comment:
Agh, my bad on the formatting somehow, this should have stayed as all one line, mind tweaking it?

Reply:
Sure thing!